### PR TITLE
Moved init of SC to constructor.

### PR DESCRIPTION
**Pull request by breakreturn**
This prepares for coming changes to states and
their meaning within SC. The creator of SC doesn't
have to call init immediately after creation in order
to do something meaningful.

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

### CODE REVIEW

File: libsoftwarecontainer/component-test/softwarecontainerlib_componenttest.cpp [hunk 1]
```diff
@@ -113,13 +113,17 @@ TEST_F(SoftwareContainerApp, DoubleIDCreatesError) {
...
-    SoftwareContainer s1(id, std::move(config));
 
-    config = createConfig();
-    SoftwareContainer s2(id, std::move(config));
+    // Containers need to be created in the same scope since they will
+    // shut down and "release" their ID when the object is destroyed
+    // when it goes out of scope.
+    ASSERT_THROW({
+        config = createConfig();
+        SoftwareContainer s1(id, std::move(config));
```
Comment (tobsan) [inline libsoftwarecontainer/component-test/softwarecontainerlib_componenttest.cpp]: Why not s1(id, createConfig())?

File: libsoftwarecontainer/include/softwarecontainer.h [hunk 2]
```diff
@@ -43,40 +43,53 @@ class Gateway;
...
     LOG_DECLARE_CLASS_CONTEXT("PCL", "SoftwareContainer library");
 
-    /*
-     * @brief creates a new SoftwareContainer instance.
+    /**
+     * @brief Creates a new SoftwareContainer instance.
      *
-     * @param id the containerID to use.
-     * @param config an object holding all needed settings for the container
+     * @param idTThe containerID to use.
```
Comment (tobsan) [inline libsoftwarecontainer/include/softwarecontainer.h]: Space here

File: libsoftwarecontainer/include/softwarecontainer.h [hunk 3]
```diff
@@ -95,28 +108,35 @@ class SoftwareContainer :
...
-    std::shared_ptr<CommandJob> createCommandJob(const std::string &command);
-
+    /**
+     *
+     */
     ReturnCode bindMount(const std::string &pathOnHost, const std::string &pathInContainer, bool readonly = true);
 
+    /**
+     *
+     */
```
Comment (tobsan) [inline libsoftwarecontainer/include/softwarecontainer.h]: Empty comments?

**Review by tobsan — CHANGES_REQUESTED** [review]
Good idea, just have some issues.

File: libsoftwarecontainer/include/softwarecontainer.h (hunk 3 above)
Comment (breakreturn) [reply libsoftwarecontainer/include/softwarecontainer.h]: Yeah, I thought it made them stand out more from all the commented ones. I could add comments to them, but good comments are sort of out of scope for the commits. Maybe I should get into that good scout feeling though, but these old sins are too annoying sometimes. I'll remove the empty comments instead (for now).

**Review by tobsan — APPROVED** [review]

**Pull request merged**